### PR TITLE
[wip] Style redbox

### PR DESCRIPTION
**Pull request by nicinabox**
screen shot 2016-11-29 at 14 15 27

Based on discussion in https://github.com/facebookincubator/create-react-app/pull/1101

### CODE REVIEW

**Comment by gaearon** [comment]
Wow, great work.
Can we make it more obvious how collapsed trace works? Those dots are a bit confusing to me.

**Comment by nicinabox** [comment]
I was thinking the same. Wasn't sure if @Timer had something in mind
already.

I was thinking something like:

```
-----[ 7 internal paths collapsed ]-----
```

**Comment by nicinabox** [comment]
This from the other thread is good too:

> 5 stack frames in ~/react were omitted.
> [Click to expand]

**Comment by Timer** [comment]
The dots need to go, they were just a placeholder I had to signify there was missing/hidden information.

I'd like to do the 7 collapsed or 5 stack frames omitted.
I can implement this.

**Comment by nicinabox** [comment]
I've got something coming up in juuuust a sec

**Comment by ekosz** [comment]
@nicinabox Looks like the target of this PR is Timer's master branch rather than the `redbox` branch. You may need to reopen this with it pointing to that branch instead

**Comment by Timer** [comment]
Should we make it click to expand or leave it a key? And does a click toggle all omitted frames or only the group selected?

**Comment by nicinabox** [comment]
@ekoz Crap. Thanks for the heads up.

**Comment by Timer** [comment]
@ekosz that's fine, I can merge (and do merge) locally.
So no worry @nicinabox.

**Comment by gaearon** [comment]
GH lets you edit branch target these days.

**Comment by nicinabox** [comment]
> Should we make it click to expand or leave it a key? And does a click toggle all omitted frames or only the group selected?

Toggling all would probably be the easiest to implement. Toggling the one is probably closer to the user's expectation.

**Comment by ekosz** [comment]
@gaearon Oh man thats awsome, sorry @nicinabox I'm behind the times

**Comment by nicinabox** [comment]
Screenshot updated.

**Comment by nicinabox** [comment]
Considering doing all function names in black. I get too glazed trying to scan lots of the gray text.

What do you all think?

Before:

screen shot 2016-11-29 at 14 32 20

After:

screen shot 2016-11-29 at 14 32 09

**Comment by Timer** [comment]
@nicinabox initially, the internal function names were lighter because they were not important -- since we hide them by default now and the user has to say "I want to see internal calls", I think it's appropriate to have them darker (a.k.a the original color / same as non-internal) so I like it.

tl;dr I like the darker function names; looks good

**Comment by Timer** [comment]
What do you think about _italicizing_ the internal names so we can distinguish them (is there such thing as a like, 50% italicize?)?

**Comment by nicinabox** [comment]
It's not _terrible_, but I find it a bit harder to grok compared to it's non-italicized cousins. Can you see the difference between the two at a glance?

screen shot 2016-11-29 at 14 46 18

**Comment by gaearon** [comment]
If anything they start to look more important so I'd rather not.

**Comment by Timer** [comment]
I'm not sure if I dig the centering ...
I think I can tell the difference, is it the 1st and the 4th that are slightly less italic?

**Comment by nicinabox** [comment]
> I'm not sure if I dig the centering

I can respect that. Normally I'm not a fan of centered text (I guess this is just a centered container. Centered text would be terrible.), but here's a little bit of the reasoning behind why I chose that:

- On wide monitors it puts the text closer to the center of the screen where we're (likely) already positioned. Less head turning, more comfortable, especially when debugging.
- There's a ton of unusable space on the right side otherwise. Only paths benefit from that long of line length.
- It also lets us center the "X internal calls collapsed", which helps remove them from the left side noise. [Here's what that looks like otherwise.](https://cloud.githubusercontent.com/assets/246280/20728828/1e65a58e-b645-11e6-9ba8-3edcf576593f.png)

**Comment by Timer** [comment]
@nicinabox I don't like it because it spans onto a new line now


Can we follow bootstrap convention?
This way it's restricted based on their screen size.

**Comment by Timer** [comment]
```css
.container {
  padding-right: 15px;
  padding-left: 15px;
  margin-right: auto;
  margin-left: auto;
}
@media (min-width: 768px) {
  .container {
    width: 750px;
  }
}
@media (min-width: 992px) {
  .container {
    width: 970px;
  }
}
@media (min-width: 1200px) {
  .container {
    width: 1170px;
  }
}
```

**Comment by nicinabox** [comment]
I thought about doing the bootstrap breakpoints originally, but it seemed like a lot of complexity in JS. I think you're going to end up with wrapping at some level regardless (either long errors or narrow windows). It's possible we should just widen the max-width to something that might accommodate most errors before they wrap.

Definitely interested in other folks thoughts on this.

**Comment by ekosz** [comment]
Shouldn't be too hard to continue to use inline styles:

```js
const containerStyle = {
  'padding-right': '15px',
  'padding-left': '15px',
  'margin-right': 'auto',
  'margin-left': 'auto',
  width: calcWidth(document.innerWidth),
};

function calcWidth(width) {
  if (width < 768) return 'auto';
  if (width >= 768 && width < 992) return '750px';
  if (width >= 992 && width < 1200) return '970px';
  return '1170px';
}
```

**Comment by Timer** [comment]
@nicinabox doesn't need to be done in js -- just a css file
See 965cf91c00c2e6f4c5d6c575f1f4233f1e3d82e7

@ekosz css is better because it will be responsive on resize instead of a one-off calculation

**Comment by Timer** [comment]
It was not my intention to attempt to step on anyone's toes by making this modification (965cf91c00c2e6f4c5d6c575f1f4233f1e3d82e7), so I hope it wasn't taken this way.

**Comment by nicinabox** [comment]
Better not to bikeshed.
